### PR TITLE
oc new-build: add --image-stream flag

### PR DESCRIPTION
**Pull request by csrwng**
Brings parity to new-build with new-app.
In new-app the --image flag is deprecated, and --image-stream is used.

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees ptal

**Comment by csrwng** [comment]
Partial fix for #7613

**Comment by bparees** [comment]
lgtm [merge]

**Comment by csrwng** [comment]
updated generated completions 
[merge]

File: pkg/cmd/cli/cmd/newbuild.go [hunk 1]
```diff
@@ -107,7 +107,8 @@ func NewCmdNewBuild(fullName string, f *clientcmd.Factory, in io.Reader, out io.
 	}
 
 	cmd.Flags().StringSliceVar(&config.SourceRepositories, "code", config.SourceRepositories, "Source code in the build configuration.")
-	cmd.Flags().StringSliceVarP(&config.ImageStreams, "image", "i", config.ImageStreams, "Name of an image stream to to use as a builder.")
+	cmd.Flags().StringSliceVarP(&config.ImageStreams, "image", "", config.ImageStreams, "Name of an image stream to to use as a builder. (deprecated)")
```
Comment (kargakis) [inline pkg/cmd/cli/cmd/newbuild.go]: There is a MarkDeprecated function from pflag that you can use. See https://github.com/openshift/origin/blob/62bcd6b8f5ee018035d1303f1db78f61f72136ef/pkg/cmd/cli/cmd/export.go#L77 for example

Comment (csrwng) [inline pkg/cmd/cli/cmd/newbuild.go]: ha, cool. thx! will add it to both new-build and new-app

**Comment by csrwng** [comment]
[merge]

**Pull request merged**